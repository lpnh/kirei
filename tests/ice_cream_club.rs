mod helper;

#[test]
fn base() {
    format_and_snapshot!("ice-cream-club/base.html");
}

#[test]
fn club() {
    format_and_snapshot!("ice-cream-club/club.html");
}

#[test]
fn flavors() {
    format_and_snapshot!("ice-cream-club/flavors.html");
}

#[test]
fn home() {
    format_and_snapshot!("ice-cream-club/home.html");
}

#[test]
fn ice_cream_card() {
    format_and_snapshot!("ice-cream-club/ice_cream_card.html");
}
