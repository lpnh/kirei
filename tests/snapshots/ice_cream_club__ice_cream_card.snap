---
source: tests/ice_cream_club.rs
description: "{#\nIce cream card component - supports various display modes\nVariables that can be set before including:\n- show_availability (default true) - show availability status\n- show_position (default false) - show position number\n- show_first_choice (default false) - highlight as first choice\n- card_style - additional CSS for the card wrapper\n- emoji_style - additional CSS for the emoji\n#}\n\n{% set show_availability = show_availability | default(default_value=true) %}\n{% set show_position = show_position | default(default_value=false) %}\n{% set show_first_choice = show_first_choice | default(default_value=false) %}\n{% set card_style = card_style | default(default_value=\"\") %}\n{% set emoji_style = emoji_style | default(default_value=\"\") %}\n\n<div class=\"card\" style=\"{{ card_style }}\">\n{# First choice badge #}\n{% if show_first_choice %}\n<div style=\"background: gold; color: #333; padding: 2px 8px; border-radius: 12px; font-size: 0.8em; margin-bottom: 8px; display: inline-block;\">🥇 First Choice</div>\n{% endif %}\n\n{# Emoji with optional styling #}\n<div style=\"text-align:center;font-size:2em;{{ emoji_style }}\">{{ ice_cream.emoji }}</div>\n\n{# Name and popularity badge #}\n<h3 style=\"margin:8px 0\">\n{{ ice_cream.name }}\n{% if ice_cream.popularity == 5 %}<span class=\"popular-badge\">Member Favorite</span>{% endif %}\n</h3>\n\n{# Description #}\n<p style=\"color:var(--muted);font-size:0.95em\">{{ ice_cream.description }}</p>\n\n{# Rating and status/position row #}\n<div style=\"display:flex;justify-content:space-between;align-items:center;margin-top:10px\">\n<div>\n{% for star in 1..=5 %}\n{% if star <= ice_cream.popularity %}⭐{% else %}☆{% endif %}\n{% endfor %}\n</div>\n<div>\n{% if show_availability %}\n{% if ice_cream.available %}\n<span style=\"color:green;font-size:0.9em\">Available today</span>\n{% else %}\n<span style=\"color:var(--muted);font-size:0.9em\">Coming soon</span>\n{% endif %}\n{% endif %}\n{% if show_position %}\n<span style=\"font-size:0.8em;color:green\">#{{ position }}</span>\n{% endif %}\n</div>\n</div>\n\n{# Seasonal badge #}\n{% if ice_cream.seasonal %}\n<div style=\"font-size:0.8em;margin-top:8px\">\n<span style=\"color:orange\">🌱 Seasonal special</span>\n</div>\n{% endif %}\n</div>\n"
---
{#
    Ice cream card component - supports various display modes
    Variables that can be set before including:
    - show_availability (default true) - show availability status
    - show_position (default false) - show position number
    - show_first_choice (default false) - highlight as first choice
    - card_style - additional CSS for the card wrapper
    - emoji_style - additional CSS for the emoji
#}
{% set show_availability = show_availability | default(default_value=true) %}
{% set show_position = show_position | default(default_value=false) %}
{% set show_first_choice = show_first_choice | default(default_value=false) %}
{% set card_style = card_style | default(default_value="") %}
{% set emoji_style = emoji_style | default(default_value="") %}
<div class="card" style="{{ card_style }}">
    {# First choice badge #}
    {% if show_first_choice %}
        <div style="background: gold; color: #333; padding: 2px 8px; border-radius: 12px; font-size: 0.8em; margin-bottom: 8px; display: inline-block;">
            🥇 First Choice
        </div>
    {% endif %}
    {# Emoji with optional styling #}
    <div style="text-align:center;font-size:2em;{{ emoji_style }}">
        {{ ice_cream.emoji }}
    </div>
    {# Name and popularity badge #}
    <h3 style="margin:8px 0">
        {{ ice_cream.name }}
        {% if ice_cream.popularity == 5 %}
            <span class="popular-badge">Member Favorite</span>
        {% endif %}
    </h3>
    {# Description #}
    <p style="color:var(--muted);font-size:0.95em">{{ ice_cream.description }}</p>
    {# Rating and status/position row #}
    <div style="display:flex;justify-content:space-between;align-items:center;margin-top:10px">
        <div>
            {% for star in 1..=5 %}
                {% if star <= ice_cream.popularity %}
                    ⭐
                {% else %}
                    ☆
                {% endif %}
            {% endfor %}
        </div>
        <div>
            {% if show_availability %}
                {% if ice_cream.available %}
                    <span style="color:green;font-size:0.9em">Available today</span>
                {% else %}
                    <span style="color:var(--muted);font-size:0.9em">Coming soon</span>
                {% endif %}
            {% endif %}
            {% if show_position %}
                <span style="font-size:0.8em;color:green">#{{ position }}</span>
            {% endif %}
        </div>
    </div>
    {# Seasonal badge #}
    {% if ice_cream.seasonal %}
        <div style="font-size:0.8em;margin-top:8px">
            <span style="color:orange">🌱 Seasonal special</span>
        </div>
    {% endif %}
</div>
